main_test.go

```diff
@@ -369,13 +369,20 @@ func runTestWithConfig(name string, t *testing.T, options compileopts.Options, c
 
 	// Create the test command, taking care of emulators etc.
 	var cmd *exec.Cmd
-	if len(spec.Emulator) == 0 {
+
+	// Probably not the best place to handle this
+	var emulator []string
+	for _, s := range spec.Emulator {
+		emulator = append(emulator, strings.ReplaceAll(s, "{root}", goenv.Get("TINYGOROOT")))
+	}
```

Comment:
Yeah, that kind of duplication doesn't look good.
What about this?

```suggestion
	config := compileopts.Config{Target: spec}
	emulator := config.Emulator()
```

A bit hacky but it works.
Also, `spec.Emulator` is also used below. It probably doesn't matter much though.

Reply:
Fixed and repushed.